src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -22,143 +32,335 @@ export const CODE_TARGET_TYPE: 'code' = 'code'
+     * from a debug-config.
+     */
+    public async launchPython(config: SamLaunchRequestArgs) {
+        //  Make a Python launch-config from the generic config.
+        const launchConfig: PythonDebugConfiguration = await pythonDebug.makePythonDebugConfig(
+            config,
+            !config.noDebug,
+            config.runtime,
+            config.handlerName,
+            config.documentUri,
+        )
+        await pythonDebug.invokePythonLambda(this.ctx, launchConfig)
+        return launchConfig
+    }
 }
 
-function createDirectInvokeSamDebugConfigurationFromTemplate(
-    resourceName: string,
-    templatePath: string
-): AwsSamDebuggerConfiguration {
-    return {
-        type: AWS_SAM_DEBUG_TYPE,
-        request: DIRECT_INVOKE_TYPE,
-        name: resourceName,
-        invokeTarget: {
-            target: TEMPLATE_TARGET_TYPE,
-            samTemplatePath: templatePath,
-            samTemplateResource: resourceName
+/**
+ * Validates debug configuration properties.
+ */
+function validateConfig(
+    folder: vscode.WorkspaceFolder | undefined,
+    config: AwsSamDebuggerConfiguration,
+): boolean {
+    const cftRegistry = CloudFormationTemplateRegistry.getRegistry()
+
+    let rv: { isValid: boolean; message?: string } = { isValid: false, message: undefined, }
+    if (!config.request) {
+        rv.message = localize(
+            'AWS.sam.debugger.missingField',
+            'Missing required field "{0}" in debug config',
+            'request'
+        )
+    } else if (!AWS_SAM_DEBUG_REQUEST_TYPES.includes(config.request)) {
+        rv.message = localize(
+            'AWS.sam.debugger.invalidRequest',
+            'Debug Configuration has an unsupported request type. Supported types: {0}',
+            AWS_SAM_DEBUG_REQUEST_TYPES.join(', '))
+    } else if (!AWS_SAM_DEBUG_TARGET_TYPES.includes(config.invokeTarget.target)) {
+        rv.message = localize(
+            'AWS.sam.debugger.invalidTarget',
+            'Debug Configuration has an unsupported target type. Supported types: {0}',
+            AWS_SAM_DEBUG_TARGET_TYPES.join(', '))
+    } else if (config.invokeTarget.target === TEMPLATE_TARGET_TYPE) {
+        let cfnTemplate
+        if (config.invokeTarget.samTemplatePath) {
+            const fullpath = tryGetAbsolutePath(folder, config.invokeTarget.samTemplatePath)
+            // Normalize to absolute path for use in the runner.
+            config.invokeTarget.samTemplatePath = fullpath
+            cfnTemplate = cftRegistry.getRegisteredTemplate(fullpath)?.template
         }
+        rv = validateTemplateConfig(config, config.invokeTarget.samTemplatePath, cfnTemplate)
+    } else if (config.invokeTarget.target === CODE_TARGET_TYPE) {
+        rv = validateCodeConfig(config)
+    }
+
+    if (!rv.isValid) {
+        vscode.window.showErrorMessage(rv.message ?? 'invalid debug-config')
+    } else if (rv.message) {
+        vscode.window.showInformationMessage(rv.message)
     }
```

Comment:
Do these need to be `await`ed/tagged with a `no-floating-promises`?

Reply:
No - awaiting showXxxMessage will cause the code to wait for the user to dismiss the notice.
Not awaiting is a fire and forget operation, which is what we want in this situation.

Reply:
My concern was more of a "will the linter let this happen" concern.

Reply:
lint passes